src/lib/WIFI/wifiJoystick.cpp

```diff
@@ -0,0 +1,147 @@
+#include "device.h"
+
+#include "wifiJoystick.h"
+
+#if defined(TARGET_TX) && defined(PLATFORM_ESP32)
+
+#include <WiFi.h>
+#include <WiFiUdp.h>
+#include "common.h"
+#include "CRSF.h"
+#include "POWERMGNT.h"
+#include "hwTimer.h"
+#include "logging.h"
+#include "options.h"
+
+
+#if defined(RADIO_SX127X)
+extern SX127xDriver Radio;
+#elif defined(RADIO_SX128X)
+extern SX1280Driver Radio;
+#endif
+
+WiFiUDP *WifiJoystick::udp = NULL;
+IPAddress WifiJoystick::remoteIP;
+uint8_t WifiJoystick::channelCount = JOYSTICK_DEFAULT_CHANNEL_COUNT;
+bool WifiJoystick::active = false;
+
+static inline uint16_t htole16(uint16_t val)
+{
+#if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+    return __builtin_bswap16(val);
+#else
+    return val;
+#endif
+}
+
+void WifiJoystick::StartJoystickService()
+{
+    if (!udp)
+    {
+        udp = new WiFiUDP();
+        udp->begin(JOYSTICK_PORT);
+    }
+}
+
+void WifiJoystick::StopJoystickService()
+{
+    active = false;
+    if (udp)
+    {
+        udp->stop();
+        delete udp;
+        udp = NULL;
+    }
+}
+
+void WifiJoystick::StartSending(const IPAddress& ip, int32_t updateInterval, uint8_t newChannelCount)
+{
+    remoteIP = ip;
+    if (!udp || active)
+    {
+        return;
+    }
+
+    // RF should already be shut down if in wifi mode
+    // Adjust the timer to run at the requested interval
+    // with a hard lower limit of 1000Hz
+    if (updateInterval == 0 || updateInterval < 1000)
+    {
+        updateInterval = JOYSTICK_DEFAULT_UPDATE_INTERVAL;
+    }
+    hwTimer::updateInterval(updateInterval);
+    CRSF::setSyncParams(updateInterval);
+
+    CRSF::RCdataCallback = &UpdateValues;
+
+    // Channel Count
+    if (newChannelCount == 0)
+    {
+        newChannelCount = JOYSTICK_DEFAULT_CHANNEL_COUNT;
+    }
+    else if (newChannelCount > 16)
+    {
+        newChannelCount = 16;
+    }
+    channelCount = newChannelCount;
+
+    active = true;
+}
+
+
+void WifiJoystick::Loop(unsigned long now)
+{
+    static unsigned long last = 0;
+    if (!udp || active)
+    {
+       return;
+    }
+
+    // Advertise the existance of the server via broadcast
+    constexpr unsigned BROADCAST_INTERVAL = 5000U;
+    if (now - last > BROADCAST_INTERVAL)
+    {
+        last = now;
+
+        struct ElrsUdpAdvertisement_s {
+            uint32_t magic;     // char[4] = ['E', 'L', 'R', 'S']
+            uint8_t version;    // JOYSTICK_VERSION
+            uint16_t port;      // JOYSTICK_PORT, little endian
+            uint8_t name_len;   // length of the device name that follows
+            char name[0];       // device name
+        } eua = {
```

Comment:
Ah TWO bugs in my code, this should be `} PACKED eua = {` because it comes out padded otherwise
```
00000000  45 4c 52 53 01 00 2a f8  08 00 00 00 52 4d 20 54  |ELRS..*.....RM T|
                                      ^^ 3 extra zero bytes here
```